content/docs/apis.md

```diff
@@ -2,57 +2,176 @@
+```rust
+// `Key` and `Value` wrap around `Vec<u8>` values.
+// This means data does not need to be UTF-8.
+let key = "TiKV".to_bytes();
+let value = "Astronaut".to_bytes();
 
-1. Install the necessary packages:
+// This creates a future that must be resolved.
+let req = client.put(key, value);
+req.wait()?;
 
-    ```shell
-    go get -v -u github.com/pingcap/tidb/store/tikv
-    ```
+let req = client.get(key);
+let result = req.wait()?;
 
-1. Import the dependency packages:
+// `Value` and `Key` deref to `Vec<u8>`.
+// They should feel natural to work with.
+assert_eq!(result, Some(value));
 
-    ```go
-    import (
-        "fmt"
-        "github.com/pingcap/tidb/config"
-        "github.com/pingcap/tidb/store/tikv"
-    )
-    ```
+let req = client.delete(key);
+req.wait()?;
 
-1. Create a raw key-value client:
+let req = client.get(key).wait()?;
+assert_eq!(result, None);
+```
 
-    ```go
-    cli, err := tikv.NewRawKVClient([]string{"192.168.199.113:2379"}, config.Security{})
-    ```
+You can also perform `scan`s:
 
-    Description of two parameters in the above command:
+```rust
+// For stability and reliability, it's good to chose a reasonable limit.
+const REASONABLE_LIMIT = 1000;
+// If you are using UTF-8,`Key` and `Value` arguments can be provided as
+// `String` or `&'static str` as well.
+const (START, END) = ("C", "F");
 
-    * `string`: a list of PD servers' addresses
-    * `config.Security`: used to establish TLS connections, usually left empty when you do not need TLS
+// Scanning can also work on an open end (Eg `START..`)
+let req = client.scan(START..END, REASONABLE_LIMIT);
+let result: Vec<Value> = req.wait()?;
+```
 
-1. Call the Raw Key-Value client methods to access the data on TiKV. The Raw Key-Value API contains the following methods, and you can also find them in the [GoDoc](https://godoc.org/github.com/pingcap/tidb/store/tikv#RawKVClient).
+These functions also have batch variants which accept sets and return `Vec<_>`s of data. These this offer considerably reduced network overhead and can result in dramatic performance increases under certain workloads.
 
-    ```go
-    type RawKVClient struct
-    func (c *RawKVClient) Close() error
-    func (c *RawKVClient) ClusterID() uint64
-    func (c *RawKVClient) Delete(key []byte) error
-    func (c *RawKVClient) Get(key []byte) ([]byte, error)
-    func (c *RawKVClient) Put(key, value []byte) error
-    func (c *RawKVClient) Scan(startKey []byte, limit int) (keys [][]byte, values [][]byte, err error)
-    ```
+For documented, tested examples of all functionalities, check the documentation of `raw::Client` in the generated Rust documentation.
 
 ## Transactional key-value API {#transactional}
+
+> The transactional API of the Rust client is incomplete. You can track the progress with [issue #15](https://github.com/tikv/client-rust/issues/15).
+
+Using a connected `transaction::Client` you can then begin a transaction:
+
+```rust
+let txn = client.begin();
```

Comment:
May as well add the line or two of boilerplate for creating the Client

Reply:
We showed them this at the top of the page and adding these ~4 lines would dilute the example to make it harder to see the way to start a transaction. I've added `// Create the client as shown above` instead. Does this work?